src/classes/STG_SettingsManager_TEST.cls

```diff
@@ -115,24 +156,29 @@ public with sharing class STG_SettingsManager_TEST {
     */
     @isTest
     private static void shouldNotRemoveAllocDefaultIfGauExist() {
+        List<Apexpages.Message> msgs;
         General_Accounting_Unit__c gau = new General_Accounting_Unit__c(Name = 'foo');
         insert gau;
 
-        STG_SettingsManager_CTRL ctrl = new STG_SettingsManager_CTRL(); 
-        
+        STG_SettingsManager_CTRL ctrl = new STG_SettingsManager_CTRL();
+
         STG_Panel.stgService.stgAllo.Default_Allocations_Enabled__c = true;
         STG_Panel.stgService.stgAllo.Default__c = gau.id;
         upsert STG_Panel.stgService.stgAllo;
 
         Test.startTest();
-        ctrl.onNpspSettingsLoad();
+        try {
+            ctrl.onNpspSettingsLoad();
+            System.assert(true, 'onNpspSettingsLoad should have thrown exception');
```

Comment:
Exception is not thrown on `ctrl.onNpspSettingsLoad();`, so the `try/catch and the `assert` statements can be removed.
Should we check if the `errorMsg` is empty?

Reply:
I agree. We're not throwing an exception in the controller. We're instantiating the exception and passing it to the error logging class. Either checking for a record in the `Error__c` object or if the `errorMsg` (or whatever the better name is) has been populated with the expected value would work.

Reply:
Thanks, @force2b ! I will check if the errorMsg has been populated with the expected value in the test. However, UTIL_UnitTestData_TEST.assertPageHasError() might not work as the page do not have error as apex messages are not set here.